src/sagas/networkSettings.js

```diff
@@ -287,6 +289,10 @@ export function* persistNetworkSettings(action) {
   // Reset network name when changing networks
   yield put(setFullNodeNetworkName(''));
 
+  // After we persist the network settings, we must create a new unleash
+  // client because the network/stage might have changed
+  yield call(monitorFeatureFlags, 0, false);
```

Comment:
Hmmm, maybe instead of calling the `monitorFeatureFlags` saga directly from this saga we should just listen to the `networkChanged` action in the featureToggle saga and react to it...